chainerx_cc/chainerx/cuda/memory_pool.h

```diff
@@ -156,6 +160,10 @@ class MemoryPool {
     cuda_internal::FreeBinsMap free_bins_;  // allocation size => cuda_internal::FreeList
     std::mutex in_use_mutex_;
     std::mutex free_bins_mutex_;
+
+    std::function<void(MemoryPool&, size_t)> malloc_preprocess_hook_;
+    std::function<void(MemoryPool&, size_t, void*)> malloc_postprocess_hook_;
+    std::function<void(MemoryPool&, void*)> free_preprocess_hook_;
```

Comment:
Now that there's only one `free` hook, `preprocess` is redandant. How about `free_hook_`? (similarly `SetFreeHook` etc.)

Reply:
Changed.